walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -64,65 +67,57 @@
         return possibleTimes;
       }
 
-      // for (Event event : requiredEvents) {
-      //   if (!possibleTimes.isEmpty()) {
-
-      //   }
-      //   else {
-      //     possibleTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false));
-      //     possibleTimes.add(TimeRange.fromStartEnd(event.getWhen().end(), TimeRange.END_OF_DAY, true));
-      //   }
-       
-      // }
-
-      // This assumes there are 2 events with required attendees 
-      if (requiredEvents.size() == 2) {
-        Iterator<Event> it = events.iterator();
-        Event firstEvent = it.next();
-        Event secondEvent = it.next();
-
-        if ((secondEvent.getWhen().start() - firstEvent.getWhen().end()) < request.getDuration()) {
-          // If one person's event fully contains another's event (nested events), there are 2 possible times: before and after the first event.
-          if (firstEvent.getWhen().end() > secondEvent.getWhen().start()
-              && firstEvent.getWhen().end() > secondEvent.getWhen().end()) {
-            possibleTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEvent.getWhen().start(), false));
-            possibleTimes.add(TimeRange.fromStartEnd(firstEvent.getWhen().end(), TimeRange.END_OF_DAY, true));
-            return possibleTimes;
+      for (int counter = 0; counter < requiredEvents.size(); counter++) {
+        Event currentEvent = requiredEvents.get(counter);
+
+        // If the array is not empty, then it means there was one event before this one. 
+        if (!possibleTimes.isEmpty()) {
+          Event previousEvent = requiredEvents.get(counter - 1);
+      
+          // Remove the last added timerange to consider the unavailability due to the current event. 
+          possibleTimes.remove(possibleTimes.size() - 1);
+
+          // If one event fully contains another (nested event), there are 2 possible times: before and after the longest event. 
+          if (previousEvent.getWhen().end() > currentEvent.getWhen().start()
```

Comment:
Nit: this logic doesn't quite say what the comment above is saying? It seems to be saying: "if the previous event ends after the current event, add the time after the previous event as a possible time". (I can see how it's related, and it's close, so I'm mostly nitpicking.)

Reply:
Oh that's true, I missed that, thanks for pointing that out Doug. I actually realised I could use the contains() function here to simplify the condition.